presto-iceberg/src/main/java/com/facebook/presto/iceberg/optimizer/IcebergEqualityDeleteAsJoin.java

```diff
@@ -383,24 +383,31 @@ private ImmutableMap<VariableReferenceExpression, ColumnHandle> createAssignment
                             if (!partitionField.transform().isIdentity()) {
                                 Type partitionFieldType = partitionField.transform().getResultType(sourceField.type());
                                 VariableReferenceExpression variableReference = variableAllocator.newVariable(partitionField.name(), toPrestoType(partitionFieldType, typeManager));
-                                IcebergColumnHandle columnHandle = new IcebergColumnHandle(
-                                        ColumnIdentity.createColumnIdentity(partitionField.name(), partitionField.fieldId(), partitionFieldType),
-                                        toPrestoType(partitionFieldType, typeManager),
-                                        Optional.empty(),
-                                        PARTITION_KEY);
-                                unselectedAssignmentsBuilder.put(variableReference, columnHandle);
+                                if (!selectedFields.contains(partitionField.fieldId())) {
+                                    IcebergColumnHandle columnHandle = new IcebergColumnHandle(
+                                            ColumnIdentity.createColumnIdentity(partitionField.name(), partitionField.fieldId(), partitionFieldType),
+                                            toPrestoType(partitionFieldType, typeManager),
+                                            Optional.empty(),
+                                            PARTITION_KEY);
+                                    unselectedAssignmentsBuilder.put(variableReference, columnHandle);
+                                    selectedFields.add(partitionField.fieldId());
+                                }
                             }
                             else if (!selectedFields.contains(sourceField.fieldId())) {
                                 unselectedAssignmentsBuilder.put(
                                         variableAllocator.newVariable(sourceField.name(), toPrestoType(sourceField.type(), typeManager)),
                                         IcebergColumnHandle.create(sourceField, typeManager, REGULAR));
+                                selectedFields.add(sourceField.fieldId());
```

Comment:
I think this and the line in the `else` case below might be the only place where there might actually be a duplicate. If you explicitly include the partition fields in the equality_delete_field_ids. 

I think that should be easy to replicate with a test.